test/functional/sqlsrv/sqlsrv_empty_result_error.phpt

```diff
@@ -0,0 +1,121 @@
+--TEST--
+Test that calling sqlsrv_next_result() on a null result set produces the correct error message. Fix for Github 507.
+--SKIPIF--
+<?php require('skipif.inc'); ?>
+--FILE--
+<?php
+require_once("MsSetup.inc");
+require_once("MsCommon.inc");
+
+$conn = sqlsrv_connect($server, array("Database"=>$databaseName, "uid"=>$uid, "pwd"=>$pwd));
+
+DropTable($conn, 'TestEmptySetTable');
+$stmt = sqlsrv_query($conn, "CREATE TABLE TestEmptySetTable ([c1] nvarchar(10),[c2] nvarchar(10))");
+$stmt = sqlsrv_query($conn, "INSERT INTO TestEmptySetTable (c1, c2) VALUES ('a', 'b')");
+
+// Create a procedure that can return a result set or can return nothing
+DropProc($conn, 'TestEmptySetProc');
+$stmt = sqlsrv_query($conn, "CREATE PROCEDURE TestEmptySetProc @a nvarchar(10), @b nvarchar(10)
+                             AS SET NOCOUNT ON
+                             BEGIN
+                                 IF @b='b'
+                                 BEGIN
+                                     SELECT 'a' as testValue
+                                 END
+                                 ELSE
+                                 BEGIN
+                                     UPDATE TestEmptySetTable SET c2 = 'c' WHERE c1 = @a
+                                 END
+                             END");
+
+// errors out when reaching the second nextRowset() call
+// returned error indicates there are no more results
+echo "Return a nonempty result set:\n";
+
+$stmt = sqlsrv_query($conn,"TestEmptySetProc @a='a', @b='b'");
+$result = sqlsrv_fetch_array($stmt);
+print_r($result);
+sqlsrv_next_result($stmt);
+$result = sqlsrv_fetch_array($stmt);
+print_r($result);
+sqlsrv_next_result($stmt);
+
+print_r(sqlsrv_errors());
+
+// errors out indicating the result set contains no fields
+echo "Return an empty result set, call nextRowset on it before fetching anything:\n";
+
+$stmt = sqlsrv_query($conn, "TestEmptySetProc @a='a', @b='c'");
+sqlsrv_next_result($stmt);
+print_r(sqlsrv_errors());
+
+// errors out indicating the result set contains no fields
+echo "Return an empty result set, call fetch on it:\n";
+
+$stmt = sqlsrv_query($conn, "TestEmptySetProc @a='a', @b='c'");
+$result = sqlsrv_fetch_array($stmt);
+print_r($result);
+print_r(sqlsrv_errors());
+
+$stmt = sqlsrv_query($conn, "DROP TABLE TestEmptySetTable");
+$stmt = sqlsrv_query($conn, "DROP PROCEDURE TestEmptySetProc");
+sqlsrv_close($conn);
+?>

```

Comment:
just for practice, probably free the statement object too before closing the connection?